### PR TITLE
Fix bugs with SVR and Polynomial kernel

### PR DESCRIPTION
**Pull request by PetrovKP**

### CODE REVIEW

**Review by owerbat — APPROVED** [review]

File: cpp/daal/src/algorithms/kernel_function/polynomial/kernel_function_polynomial_dense_default_impl.i [hunk 1]
```diff
@@ -185,12 +185,13 @@ services::Status KernelImplPolynomial<defaultDense, algorithmFPType, cpu>::compu
...
-                        if (par->degree != 0)
+                        if (degree != 0)
                         {
                             dataR[i * nVectors2 + j] += shift;
-                            for (size_t k = 0; k < par->degree - 1; ++k)
+                            const algorithmFPType factor = dataR[i * nVectors2 + j];
+                            for (size_t k = 0; k < degree - 1; ++k)
                             {
-                                dataR[i * nVectors2 + j] *= dataR[i * nVectors2 + j];
+                                dataR[i * nVectors2 + j] *= factor;
```
Comment (SmirnovEgorRu) [inline cpp/daal/src/algorithms/kernel_function/polynomial/kernel_function_polynomial_dense_default_impl.i:194]: Need to add tests to catch the problem in the future.

Comment (PetrovKP) [reply cpp/daal/src/algorithms/kernel_function/polynomial/kernel_function_polynomial_dense_default_impl.i:194]: I have tests on python side for poly kernel: https://github.com/intel/scikit-learn-intelex/pull/572/files#diff-c1e6282c1d0df1d714b95b9f994fc5efcb3d2cfd6635ff001e2c6e0d6d0f7d72R106

Comment (SmirnovEgorRu) [reply cpp/daal/src/algorithms/kernel_function/polynomial/kernel_function_polynomial_dense_default_impl.i:194]: I mean C++ tests.
Just extend this https://github.com/oneapi-src/oneDAL/blob/4e0570d6ce8e43843e3b5547408c8963689c941b/cpp/oneapi/dal/algo/polynomial_kernel/test/batch.cpp#L145
to degrees more than 2.

**Pull request merged**